pkg/cosmos/client/test_helpers.go

```diff
@@ -205,3 +214,16 @@ func mustRandomPort() int {
 	}
 	return int(r.Int64() + 1024)
 }
+
+// awaitTxCommitted waits for a transaction to be committed on chain and returns the tx receipt
+func awaitTxCommitted(t *testing.T, tc *Client, txHash string) (response *txtypes.GetTxResponse, success bool) {
+	for i := 0; i < 9; i++ { // max poll attempts to wait for tx commitment
+		txReceipt, err := tc.Tx(txHash)
+		if err == nil {
+			return txReceipt, true
+		}
+		t.Logf("tx not committed yet (attempt %d)", i+1)
+		time.Sleep(time.Second * 1) // TODO: configure dynamically based on block times
+	}
+	return nil, false
+}
```

Comment:
Avg observed tx confirmation time was ~5 seconds, this polls for up to 10 seconds (arbitrary). We could set this to the default timeout defined in the Client (30 seconds) however in a testing environment 10 seconds should be sufficient, and increasing the wait duration may increase the duration of tests even more - currently the go unit tests with race detector enabled run for around 560-580 seconds, almost hitting the timeout at 600 seconds.